D/Dyninst/build_tarballs.jl

```diff
@@ -0,0 +1,161 @@
+# ldd /workspace/destdir/lib/libtbb.so.12 || true
+# ldd /workspace/destdir/lib/libstdc++.so.6 || true
+# ldd /lib64/libm.so.6 || true
+# ldd /workspace/destdir/lib/libgomp.so.1 || true
+# ldd /workspace/destdir/lib/libgcc_s.so.1 || true
+# ldd /lib64/libc.so.6 || true
+# ldd /lib64/librt.so.1 || true
+# ldd /lib64/libdl.so.2 || true
+
+cat >dlopen.c <<EOF
+#include <dlfcn.h>
+#include <stdio.h>
+int main(int *argc, char *argv[]) {
+  // void *ptr = dlopen(argv[1], RTLD_NOW);
+  void *ptr = dlopen(argv[1], RTLD_LAZY | RTLD_DEEPBIND);
+  printf("ptr=%p\n", ptr);
+  return 0;
+}
+EOF
+cc -o dlopen dlopen.c -ldl
+
+./dlopen $libdir/libcommon.so
+./dlopen $libdir/libdynC_API.so
+./dlopen $libdir/libdynDwarf.so
+./dlopen $libdir/libdynElf.so
+./dlopen $libdir/libdyninstAPI.so
+./dlopen $libdir/libdyninstAPI_RT.so
+./dlopen $libdir/libinstructionAPI.so
+./dlopen $libdir/libparseAPI.so
+./dlopen $libdir/libpatchAPI.so
+./dlopen $libdir/libpcontrol.so
+./dlopen $libdir/libstackwalk.so
+./dlopen $libdir/libsymLite.so
+./dlopen $libdir/libsymtabAPI.so
+
+#TODO
+# exit 1
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+platforms = expand_cxxstring_abis(platforms)
+
+# Binutils and Elfutils require Linux
+# TODO: We should be able to build on Windows without Binutils and Elfutils
+filter!(Sys.islinux, platforms)
+# cmake fails with "unknown platform"
+filter!(p -> arch(p) ∉ ["armv6l", "armv7l"], platforms)
+# linking fails with "undefined reference to `_r_debug'"
+# TODO: Would this be fixed by linking against `libdl`?
+filter!(p -> libc(p) ≠ "musl", platforms)
+
+# TODO: For debugging:
+filter!(p -> arch(p) == "x86_64" && Sys.islinux(p) && libc(p) == "glibc" && cxxstring_abi(p) == "cxx11",  platforms)
+
+# The products that we will ensure are always built
+products = [
+    ExecutableProduct("parseThat", :parseThat),
+    LibraryProduct("libcommon", :libcommon),
+    LibraryProduct("libdynC_API", :libdynC_API),
+    LibraryProduct("libdynDwarf", :libdynDwarf),
+    LibraryProduct("libdynElf", :libdynElf),
+    LibraryProduct("libdyninstAPI", :libdyninstAPI),
+    LibraryProduct("libdyninstAPI_RT", :libdyninstAPI_RT),
+    LibraryProduct("libinstructionAPI", :libinstructionAPI),
+    LibraryProduct("libparseAPI", :libparseAPI),
+    LibraryProduct("libpatchAPI", :libpatchAPI),
+    LibraryProduct("libpcontrol", :libpcontrol),
+    LibraryProduct("libstackwalk", :libstackwalk),
+    LibraryProduct("libsymLite", :libsymLite),
+    LibraryProduct("libsymtabAPI", :libsymtabAPI),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    BuildDependency("CMake_jll"),
+    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae")),
+    # We need at least v2.41 of Binutils_jll to get an `-fPIC` version of `libiberty.a`
+    Dependency("Binutils_jll"; compat="2.41"),
```

Comment:
Don't you need a binutils for the host, rather than the target?

Reply:
Binutils provides a static library `libiberty.a` that is required here (for the target). I don't care about the utilities themselves. Do you think this would overwrite something?

Anyway, the current problem I'm seeing occurs when host = target, so I hope I can address the Binutils problem later, if there is one.

Reply:
Ok, then it has to be a `BuildDependency` if you only need to grab a static library.

Reply:
Did that.